pkg/solana/logpoller/parser.go

```diff
@@ -1,5 +1,413 @@
 package logpoller
 
+import (
+	"errors"
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gagliardetto/solana-go"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+)
+
+const (
+	blockFieldName     = "block_number"
+	chainIDFieldName   = "chain_id"
+	timestampFieldName = "block_timestamp"
+	txHashFieldName    = "tx_hash"
+	addressFieldName   = "address"
+	eventSigFieldName  = "event_sig"
+	defaultSort        = "block_number ASC, log_index ASC"
+)
+
 var (
-	logsFields = [...]string{"id", "filter_id", "chain_id", "log_index", "block_hash", "block_number", "block_timestamp", "address", "event_sig", "subkey_values", "tx_hash", "data", "created_at", "expires_at", "sequence_num"}
+	ErrInvalidComparator   = errors.New("invalid comparison operator")
+	ErrInvalidConfidence   = errors.New("invalid confidence level; solana only supports finalized")
+	ErrInvalidCursorDir    = errors.New("invalid cursor direction")
+	ErrInvalidCursorFormat = errors.New("invalid cursor format")
+	ErrInvalidSortDir      = errors.New("invalid sort direction")
+	ErrInvalidSortType     = errors.New("invalid sort by type")
```

Comment:
Good call on defining these all together! This will help later with abstraction.